src/components/InteractiveStepSubHeader.tsx

```diff
@@ -0,0 +1,106 @@
+import type {ForwardedRef} from 'react';
+import React, {forwardRef, useImperativeHandle, useState} from 'react';
+import {View} from 'react-native';
+import useThemeStyles from '@hooks/useThemeStyles';
+import colors from '@styles/theme/colors';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import PressableWithFeedback from './Pressable/PressableWithFeedback';
+import Text from './Text';
+
+type InteractiveStepSubHeaderProps = {
+    /** List of the Route Name to navigate when the step is selected */
+    stepNames: readonly string[];
+
+    /** Function to call when a step is selected */
+    onStepSelected?: (stepName: string) => void;
+
+    /** The index of the step to start with */
+    startStepIndex?: number;
+};
+
+type InteractiveStepSubHeaderHandle = {
+    /** Move to the next step */
+    moveNext: () => void;
+};
+
+const MIN_AMOUNT_FOR_EXPANDING = 3;
+const MIN_AMOUNT_OF_STEPS = 2;
+
+function InteractiveStepSubHeader({stepNames, startStepIndex = 0, onStepSelected}: InteractiveStepSubHeaderProps, ref: ForwardedRef<InteractiveStepSubHeaderHandle>) {
+    const styles = useThemeStyles();
+
+    if (stepNames.length < MIN_AMOUNT_OF_STEPS) {
+        throw new Error(`stepNames list must have at least ${MIN_AMOUNT_OF_STEPS} elements.`);
+    }
+
+    const [currentStep, setCurrentStep] = useState(startStepIndex);
+    useImperativeHandle(
+        ref,
+        () => ({
+            moveNext: () => {
+                setCurrentStep((actualStep) => actualStep + 1);
+            },
+        }),
+        [],
+    );
+
+    const amountOfUnions = stepNames.length - 1;
+
+    return (
+        <View style={[styles.interactiveStepHeaderContainer, {minWidth: stepNames.length < MIN_AMOUNT_FOR_EXPANDING ? '60%' : '100%'}]}>
```

Comment:
Shouldn't this part be extracted? 
```ts
{minWidth: stepNames.length < MIN_AMOUNT_FOR_EXPANDING ? '60%' : '100%'}]}
```

Reply:
What do You mean, save this logic into  variable?

Reply:
@blazejkustra  friendly bump